File: backend/src/main/java/ProgramaClip/ClipMoneyAPI/entidad/TipoPlazoFijo.java
package ProgramaClip.ClipMoneyAPI.entidad;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import com.sun.istack.NotNull;

import lombok.Data;

@Entity
@Data
public class TipoPlazoFijo {
	@Id
	@GeneratedValue
	private Integer id;
	@NotNull
	private String nombre;
	@NotNull
	private Integer plazoMinimoDias;
	@NotNull
	private Float interesAnual;
	@NotNull
	private Float montoMinimo;
	@ManyToOne
	@NotNull
	private Moneda moneda;
}

File: Models/TipoPlazoFijo.cs
namespace proyecto_clip_money_e_proyecto_clip.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("TiposPlazosFijos")]
    public partial class TipoPlazoFijo
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public TipoPlazoFijo()
        {
            PlazosFijos = new HashSet<PlazoFijo>();
        }

        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Nombre { get; set; }

        public int PlazoMinimoDias { get; set; }

        public double InteresAnual { get; set; }

        public double MontoMinimo { get; set; }

        public int MonedaId { get; set; }

        public virtual Moneda Moneda { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<PlazoFijo> PlazosFijos { get; set; }
    }
}
